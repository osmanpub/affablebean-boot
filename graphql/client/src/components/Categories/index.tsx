import { useQuery } from "@apollo/react-hooks";
import React from "react";
import { getId } from "../../helpers/utils";
import { CategoryState } from "../../interfaces/categories";
import { GET_CATEGORIES } from "../../queries";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome,
} from "./Categories.styles";

function Categories() {
  const { data } = useQuery(GET_CATEGORIES);

  if (!data) {
    return null;
  }

  const { categories } = data;

  if (!categories || !categories.length) {
    return null;
  }

  const categoriesList = categories.map((category: CategoryState) => (
    <Category key={getId(category)} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting data-cy="home-welcome" data-testid="home-welcome">
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categoriesList}</CategoriesRight>
    </div>
  );
}

export default Categories;
