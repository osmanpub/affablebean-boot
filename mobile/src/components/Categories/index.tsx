import React from 'react';
import {FlatList, View} from 'react-native';
import {connect} from 'react-redux';
import {getId} from '../../helpers/utils';
import {
  Categories as CategoriesState,
  CategoryState,
} from '../../interfaces/categories';
import {RootState} from '../../redux';
import Category from '../Category';

type Props = {
  categories: CategoriesState;
  setCategoryProduct: Function;
};

function Categories(props: Props) {
  const {categories, setCategoryProduct} = props;
  const {items} = categories;

  if (items.length === 0) {
    return null;
  }

  return (
    <View
      style={{
        alignItems: 'center',
      }}>
      <FlatList
        data={items}
        keyExtractor={(item: CategoryState) => getId(item).toString()}
        renderItem={({item}) => (
          <Category category={item} setCategoryProduct={setCategoryProduct} />
        )}
      />
    </View>
  );
}

const mapStateToProps = (state: RootState) => ({
  categories: state.categories,
});

export default connect(mapStateToProps)(Categories);
