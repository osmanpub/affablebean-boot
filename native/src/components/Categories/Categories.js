import React from "react";
import PropTypes from "prop-types";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

export function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map(category => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}

Categories.propTypes = {
  categories: PropTypes.array.isRequired
};
