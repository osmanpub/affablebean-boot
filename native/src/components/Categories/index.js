import React from "react";
import { FlatList, View } from "react-native";
import PropTypes from "prop-types";
import Category from "../Category";

export default function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  return (
    <View
      style={{
        alignItems: "center",
        flex: 1
      }}
    >
      <FlatList
        data={props.categories}
        keyExtractor={(item, index) => item._links.self.href}
        renderItem={({ item }) => <Category category={item} />}
      />
    </View>
  );
}

Categories.propTypes = {
  categories: PropTypes.array.isRequired
};
