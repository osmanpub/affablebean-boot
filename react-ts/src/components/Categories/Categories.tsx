import React from "react";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";
import { Category as ICategory } from "../../interfaces/categories";
import { CategoriesProps } from "../../interfaces/props";

export function Categories(props: CategoriesProps) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map((category: ICategory) => (
    <Category {...category} key={category.id} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>
      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}
