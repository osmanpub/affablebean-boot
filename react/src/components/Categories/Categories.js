import React from "react";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

export function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map(category => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}
