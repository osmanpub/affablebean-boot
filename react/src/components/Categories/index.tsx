import React from "react";
import { CategoryState } from "../../redux/categories";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

type Props = {
  categories: Array<CategoryState>;
};

export default function Categories(props: Props) {
  const { categories } = props;

  if (!categories || categories.length === 0) {
    return null;
  }

  const categoriesList = categories.map((category: CategoryState) => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categoriesList}</CategoriesRight>
    </div>
  );
}
