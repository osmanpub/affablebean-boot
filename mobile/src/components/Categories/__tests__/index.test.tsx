import {render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Categories from '..';

const mockStore = configureStore([]);

const initialState = {
  categories: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        id: 1,
        name: 'dairy',
      },
      {
        id: 2,
        name: 'meats',
      },
      {
        id: 3,
        name: 'bakery',
      },
      {
        id: 4,
        name: 'fruit & veg',
      },
      {
        id: 5,
        name: 'cereals',
      },
      {
        id: 6,
        name: 'drinks',
      },
    ],
  },
};

const store = mockStore(initialState);

const categories = (
  <Provider store={store}>
    <Categories setCategoryProduct={jest.fn()} />
  </Provider>
);

describe('<Categories />', () => {
  it('renders correctly', () => {
    const component = renderer.create(categories).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('show six categories', () => {
    const {getAllByTestId} = render(categories);
    const boxes = getAllByTestId(/categoryBox/i);
    expect(boxes).toHaveLength(6);
  });
});
