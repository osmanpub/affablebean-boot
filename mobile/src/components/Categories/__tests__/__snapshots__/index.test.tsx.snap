// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <ScrollView
    ListFooterComponent={
      <View
        style={
          Object {
            "padding": 64,
          }
        }
      />
    }
    data={
      Array [
        Object {
          "id": 1,
          "name": "dairy",
        },
        Object {
          "id": 2,
          "name": "meats",
        },
        Object {
          "id": 3,
          "name": "bakery",
        },
        Object {
          "id": 4,
          "name": "fruit & veg",
        },
        Object {
          "id": 5,
          "name": "cereals",
        },
        Object {
          "id": 6,
          "name": "drinks",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          dairy
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/dairy.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          meats
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/meats.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          bakery
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/bakery.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          fruit & veg
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/fruitveg.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          cereals
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/cereals.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 24,
          }
        }
      >
        <Text>
          drinks
        </Text>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/categories/drinks.jpg",
              }
            }
          />
        </TouchableWithoutFeedback>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "padding": 64,
          }
        }
      />
    </View>
  </ScrollView>
</View>
`;
